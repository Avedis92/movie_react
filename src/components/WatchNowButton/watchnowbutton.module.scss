@import "../sasscommonmixxins/button.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.watchNowContainer {
  height: 100%;
  width: 30%;
  :nth-child(1) {
    @include styleButtons(100%, 100%, white, rgb(0, 195, 255), 10px, none);
    cursor: pointer;
    transition: 500ms;
    &:hover {
      transform: scale(105%);
    }
  }
  @include mobile {
    width: 40%;
  }
}
