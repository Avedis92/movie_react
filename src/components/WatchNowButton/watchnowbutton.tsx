import styles from './watchnowbutton.module.scss';

interface WatchNow {
    idMovie:number,
    openMovieTrailer(movieId:number):void
}

function WatchNowButton({ idMovie, openMovieTrailer }:WatchNow) {
  const { watchNowContainer } = styles;
  const handleClick = (e:React.MouseEvent<HTMLButtonElement>) => {
    openMovieTrailer(Number((e.target as HTMLButtonElement).id));
  };
  return (
    <div className={watchNowContainer}>
      <button id={String(idMovie)} name="watchNowButton" type="button" onClick={handleClick}>Watch Now</button>
    </div>
  );
}
export default WatchNowButton;
