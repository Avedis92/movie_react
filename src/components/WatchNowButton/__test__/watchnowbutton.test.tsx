import { render } from '@testing-library/react';
import WatchNowButton from '../watchnowbutton';

test('Create a snapshot test', () => {
  const {asFragment} = render(<WatchNowButton
    idMovie={550}
    openMovieTrailer={() => null}
  />);
  expect(asFragment()).toMatchSnapshot();
});
