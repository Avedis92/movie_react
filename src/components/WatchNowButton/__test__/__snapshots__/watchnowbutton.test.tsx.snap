// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="watchNowContainer"
  >
    <button
      id="550"
      name="watchNowButton"
      type="button"
    >
      Watch Now
    </button>
  </div>
</DocumentFragment>
`;
